pkg/runner/runner.go

```diff
@@ -140,6 +142,17 @@ func Run(cfg *Config) error {
 }
 
 func CreateClient(cfg *Config, promMetrics *api.Metrics) (*pgclient.Client, error) {
+	//The TimescaleDB migration has to happen before other connections
+	//are open. Also, it has to happen as the first command on a connection
+	//thus, we cannot rely on a migration lock here. Instead we assume
+	//that upgrading TimescaleDB will not break existing connectors.
```

Comment:
I'm confused by this. IIUC one of two things must be true

1. The TimescaleDB migration must be the only connection open on the DB. In this case that guarantee will ensure that there are no other connectors.
2. The migration only has to be the first command on its connection, in which case we should be able to hold the lock on a different connection

If neither of these are the case, I'd be a little concerned; while I don't think the DB will have many breaking changes after 2.0 at minimum that upgrade is something we need to be careful about.

Reply:
2 is true